From f8d41f8f7b2fa6b2a4f7546efc0dc5dde558e31e Mon Sep 17 00:00:00 2001
From: Rahul Lakkireddy <rahul.lakkireddy@chelsio.com>
Date: Fri, 29 Sep 2017 12:07:26 +0530
Subject: [PATCH 31/31] iw_cxgb4: put ep reference in pass_accept_req()

commit 3d318605f5e32ff44fb290d9b67573b34213c4c8
Author: Steve Wise <swise@opengridcomputing.com>
Date:   Wed Sep 13 09:52:32 2017 -0700

    iw_cxgb4: put ep reference in pass_accept_req()

    The listening endpoint should always be dereferenced at the end of
    pass_accept_req().

    Fixes: f86fac79afec ("RDMA/iw_cxgb4: atomic find and reference for listening endpoints")

    Cc: stable@vger.kernel.org
    Signed-off-by: Steve Wise <swise@opengridcomputing.com>
    Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/infiniband/hw/cxgb4/cm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/infiniband/hw/cxgb4/cm.c b/drivers/infiniband/hw/cxgb4/cm.c
index 0c4e86b2..e5eccc31 100644
--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -2644,9 +2644,9 @@ static int pass_accept_req(struct c4iw_dev *dev, struct sk_buff *skb)
 	goto out;
 reject:
 	reject_cr(dev, hwtid, skb);
+out:
 	if (parent_ep)
 		c4iw_put_ep(&parent_ep->com);
-out:
 	return 0;
 }
 
-- 
2.14.1

